### DISCUSSION TITLE
Covariance and Contravariance in C#

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#55] [General]
# Covariance and Contravariance in C#

Let's talk about Contravariance and Covariance in C# using .NET Framework examples!
Contravariance and covariance are essential concepts in C# when dealing with generics, enabling us to have more flexibility when assigning generic types. So let's have examples straight from the framework itself!
Also, we will go a bit deeper and talk what some differences between generic constraints and things like Contravariance are.

https://steven-giesel.com/blogPost/cda93276-c9ab-42c3-8288-a9b3e5c8aa5c

**Comment by parasparmar** [comment]
Nicely put, short sharp but extremely grateful to learn something in such a short time. 
Respects... @steven

**Reply by linkdotnet** [reply]
Hey @parasparmar thanks for the kind words and I appreciate it very much! If you have any further topics you like to read, just let me know!
